main.py

```diff
@@ -144,7 +144,13 @@ def list_circuits(self):
         archived = request.args.get("archived", "false").lower()
         archived_to_optional = {"null": None, "true": True, "false": False}
         archived = archived_to_optional.get(archived, False)
-        circuits = self.mongo_controller.get_circuits(archived=archived)
+        arg_k = request.args.get("metadata_key")
+        arg_v = request.args.get("metadata_value")
+        if arg_k:
```

Comment:
Let's simplify this logic here, once args have been parsed with their default values let's pass them all to `get_circuits`, that way the testability of the conditionals is moved to controllers that are easier to cover. 

You can have a preparsing here mapping the metadata_key and metadata_value, and possibly also use your suggestion of `metadata.*=value` in the keys and they find all args that start with `"metadata."`.

Reply:
Changed to suggestion